File: src/main/java/ec/edu/grupo3/model/MassConverterModel.java
package ec.edu.grupo3.model;

public class MassConverterModel {

    public double convert(double value, MassUnit from, MassUnit to) {
        double valueInKg = switch (from) {
            case KILOGRAM -> value;
            case GRAM -> value / 1000.0;
            case POUND -> value * 0.45359237;
            case OUNCE -> value * 0.028349523125;
        };

        return switch (to) {
            case KILOGRAM -> valueInKg;
            case GRAM -> valueInKg * 1000.0;
            case POUND -> valueInKg / 0.45359237;
            case OUNCE -> valueInKg / 0.028349523125;
        };
    }
}

File: dotnet-wcf/ConversionApp/Models/MassConverterModel.cs
using System;

namespace ConversionApp.Models
{
    public class MassConverterModel
    {
        public double Convert(double value, MassUnit from, MassUnit to)
        {
            double valueInKg = from switch
            {
                MassUnit.KILOGRAM => value,
                MassUnit.GRAM => value / 1000.0,
                MassUnit.POUND => value * 0.45359237,
                MassUnit.OUNCE => value * 0.028349523125,
                _ => throw new ArgumentException("Unidad de masa no soportada")
            };

            return to switch
            {
                MassUnit.KILOGRAM => valueInKg,
                MassUnit.GRAM => valueInKg * 1000.0,
                MassUnit.POUND => valueInKg / 0.45359237,
                MassUnit.OUNCE => valueInKg / 0.028349523125,
                _ => throw new ArgumentException("Unidad de masa no soportada")
            };
        }
    }
}
